tests/keras/utils/data_utils_test.py

```diff
@@ -64,5 +77,154 @@ def test_data_utils(in_tmpdir):
     os.remove('test.txt')
     os.remove('test.zip')
 
+
+"""Enqueuers Tests"""
+
+
+class threadsafe_iter:
+    """Takes an iterator/generator and makes it thread-safe by
+    serializing call to the `next` method of given iterator/generator.
+    """
+
+    def __init__(self, it):
+        self.it = it
+        self.lock = threading.Lock()
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        return self.next()
+
+    def next(self):
+        with self.lock:
+            return next(self.it)
+
+
+def threadsafe_generator(f):
+    """A decorator that takes a generator function and makes it thread-safe.
+    """
+
+    def g(*a, **kw):
+        return threadsafe_iter(f(*a, **kw))
+
+    return g
+
+
+class TestSequence(Sequence):
+    def __init__(self, shape):
+        self.shape = shape
+
+    def __getitem__(self, item):
+        return np.ones(self.shape, dtype=np.uint8) * item
+
+    def __len__(self):
+        return 100
+
+
+class FaultSequence(Sequence):
+    def __getitem__(self, item):
+        raise IndexError(item, 'is not present')
+
+    def __len__(self):
+        return 100
+
+
+@threadsafe_generator
+def create_generator_from_dataset_threads(ds):
```

Comment:
sorry I was unclear, I meant create_generator_from_dataset_threads => create_generator_from_sequence_threads and other instances of `dataset`